src/Microsoft.Identity.Client/ClientApplicationBase.cs

```diff
@@ -115,7 +115,7 @@ public IEnumerable<User> Users
                     return new List<User>();
                 }
 
-                return UserTokenCache.GetUsers(ClientId);
+                return UserTokenCache.GetUsers(new Uri(Authority).Host);
```

Comment:
Client id is still being used. There was no need to pass it in the function as it was present as a class property. We just added an additional constraint.